src/router.js

```diff
@@ -139,6 +139,88 @@ router.put('/registrations/:id', async (request, response) => {
   }
 });
 
+router.get('/returns', async (request, response) => {
+  try {
+    const returns = await Return.findAll();
+    if (returns) {
+      // If they are, send back the finalised registrations.
+      response.status(200).send(returns);
+    } else {
+      response.status(404).send({message: `No returns found.`});
+      return;
+    }
+  } catch (error) {
+    response.status(500).send({error});
+  }
+});
+
+router.get('/registrations/:id/return', async (request, response) => {
+  try {
+    // Try to parse the incoming ID to make sure it's really a number.
+    const existingId = Number(request.params.id);
+    if (isNaN(existingId)) {
+      response.status(404).send({message: `Registration ${request.params.id} not valid.`});
+      return;
+    }
+
+    // Check if there's a registration allocated at the specified ID.
+    const existingReg = await Registration.findOne(existingId);
+    if (existingReg === undefined || existingReg === null) {
+      response.status(404).send({message: `Registration ${existingId} not allocated.`});
+      return;
+    }
+
+    const returns = await Return.findRegReturns(existingId);
+
+    if (returns) {
+      // If they are, send back the finalised registration.
+      response.status(200).send(returns);
+    } else {
+      response.status(404).send({message: `Registration ${request.params.id} not valid.`});
+      return;
+    }
+  } catch (error) {
+    response.status(500).send({error});
+  }
+});
+
+router.get('/registrations/:id/return/:returnId', async (request, response) => {
+  try {
+    // Try to parse the incoming ID to make sure it's really a number.
+    const existingId = Number(request.params.id);
+    if (isNaN(existingId)) {
+      response.status(404).send({message: `Registration ${request.params.id} not valid.`});
+      return;
+    }
+
+    // Check if there's a registration allocated at the specified ID.
+    const existingReg = await Registration.findOne(existingId);
+    if (existingReg === undefined || existingReg === null) {
+      response.status(404).send({message: `Registration ${existingId} not allocated.`});
+      return;
+    }
+
+    // Try to parse the incoming ID to make sure it's really a number.
+    const existingReturnId = Number(request.params.returnId);
+    if (isNaN(existingReturnId)) {
+      response.status(404).send({message: `Return ${request.params.returnId} not valid.`});
+      return;
+    }
+
+    // Check if there's return allocated at the specified ID.
+    const existingReturn = await Return.findOne(existingReturnId);
+    if (existingReturn === undefined || existingReturn === null) {
+      response.status(404).send({message: `Return ${existingReturnId} not allocated.`});
+      return;
+    }
+
+    // If they are, send back the finalised return.
+    response.status(200).send(existingReturn);
```

Comment:
You've checked that the registration exists and the return exists, but not that the return belongs to the registration...

i.e.

```json
[{
  "registration": 1,
  "returns": [
    { "return": 2 },
    { "return": 3 }
  ]
}, {
  "registration": 4,
  "returns": [
    { "return": 5 },
    { "return": 6 }
  ]
}]
```

should yield only...

- `/registration/1/return/2`
- `/registration/1/return/3`
- `/registration/4/return/5`
- `/registration/4/return/6`

...but your code allows...

- `/registration/1/return/6`
- `/registration/4/return/2`
- etc.